js/packages/web/src/views/auctionCreate/reviewStep.tsx

```diff
@@ -1,18 +1,82 @@
-import { Connection } from '@solana/web3.js';
+import {
+  MAX_AUCTION_DATA_EXTENDED_SIZE,
+  MAX_EXTERNAL_ACCOUNT_SIZE,
+  MAX_VAULT_SIZE,
+} from '@oyster/common';
+import { Connection, LAMPORTS_PER_SOL } from '@solana/web3.js';
 import { Button, Col, Divider, Row, Space, Statistic } from 'antd';
 import moment from 'moment';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { AuctionCategory, AuctionState } from '.';
+import { AmountLabel } from '../../components/AmountLabel';
 import { ArtCard } from '../../components/ArtCard';
 
+// TODO: Move to common
+const BASE_SAFETY_CONFIG_SIZE =
+  1 + // Key
+  32 + // auction manager lookup
+  8 + // order
+  1 + // winning config type
+  1 + // amount tuple type
+  1 + // length tuple type
+  4 + // u32 for amount range vec
+  1 + // participation config option
+  1 + // winning constraint
+  1 + // non winning constraint
+  9 + // fixed price + option of it
+  1 + // participation state option
+  8 + // collected to accept payment
+  20; // padding
+
+const calculateAuctionCreationCost = async (connection: Connection) => {
+  const maxVaultSize =
+    connection.getMinimumBalanceForRentExemption(MAX_VAULT_SIZE);
+  const maxExternalAccountSize = connection.getMinimumBalanceForRentExemption(
+    MAX_EXTERNAL_ACCOUNT_SIZE,
+  );
+  const auctionCosts = connection.getMinimumBalanceForRentExemption(
+    MAX_AUCTION_DATA_EXTENDED_SIZE,
+  );
+  // SAFETY DEPOSIT BOX COSTS
+  const safetyDepositBoxCosts = connection.getMinimumBalanceForRentExemption(
+    BASE_SAFETY_CONFIG_SIZE,
+  );
+  // There are probably a few more costs to account for, but this is a good start
+  const allValues = await Promise.all([
+    maxVaultSize,
+    maxExternalAccountSize,
+    auctionCosts,
+    safetyDepositBoxCosts,
+  ]);
+  return allValues.reduce((acc, curr) => acc + curr, 0);
+};
+
 export const ReviewStep = (props: {
   confirm: () => void;
   attributes: AuctionState;
   setAttributes: Function;
   connection: Connection;
 }) => {
+  const [cost, setCost] = useState(0);
   const item = props.attributes.items?.[0];
 
+  useEffect(() => {
```

Comment:
Let's get a loading state in for the cost calculating. 

Toggle load off after all the responses are back.